lib/common_fw/common_fw.cpp

```diff
@@ -149,9 +151,16 @@ uint16_t Fletcher16(uint8_t *data, int count) {
   return (sum2 << 8) | sum1;
 }
 
-void debug(String str, int flag){
-  if (flag == 0){
-    Serial.println(str);
+void debug(std::string str){
+  #ifdef DEBUG
+    Serial.write(str.c_str());
     Serial.flush();
-  }
+  #endif
+}
+
+void debug(uint8_t *bytes, int count){
+  #ifdef DEBUG
+    Serial.write(bytes, count);
+    Serial.flush();
+  #endif
```

Comment:
this might be more useful if you print out the actual hex representation of the packet and not just the packet, because it will be a bunch of junk in the serial console.